torch/_inductor/codegen/cuda/cutlass_utils.py

```diff
@@ -293,13 +293,32 @@ def get_max_alignment(inductor_layout: Layout) -> int:
     def is_static_int(number):
         return isinstance(number, (int, sympy.Integer))
 
-    if is_static_int(size[-1]) and is_static_int(offset):
-        alignments = get_alignments(dtype)
-        for alignment in alignments:
-            if int(size[-1]) % alignment == 0 and int(offset) % alignment == 0:
+    try:
+        contiguous_dim = inductor_layout.stride.index(1)
+        if is_static_int(size[contiguous_dim]) and is_static_int(offset):
+            alignments = get_alignments(dtype)
+            for alignment in alignments:
+                if (
+                    int(size[contiguous_dim]) % alignment != 0
+                    or int(offset) % alignment != 0
+                ):
+                    continue
+                do_continue = False
+                for dim in range(len(size)):
+                    # All non-contiguous strides must be aligned
+                    if dim == contiguous_dim:
+                        continue
+                    dim_stride = int(inductor_layout.stride[dim])
+                    if dim_stride % alignment != 0:
+                        do_continue = True
+                        break
+                if do_continue:
+                    continue
```

Comment:
nit: I think an `if any(...): continue` expression here would be more readable / less imperative